MantidQt/CustomInterfaces/src/Reflectometry/QtReflRunsTabView.cpp

```diff
@@ -46,13 +46,13 @@ void QtReflMainView::initLayout() {
   ui.splitterTables->setStretchFactor(0, 0);
   ui.splitterTables->setStretchFactor(1, 1);
 
+  // Create the DataProcessor presenter
   ReflGenericDataProcessorPresenterFactory presenterFactory;
```

Comment:
OK, I can inject the factory into the constructor of `QtReflRunsTabView`, but `QtReflMainWindowView` will still need to create an object of type `ReflGenericDataProcessorPresenterFactory` and do something like:

```
void QtReflMainWindowView::initLayout() {
  m_ui.setupUi(this);

  ReflGenericDataProcessorPresenterFactory presenterFactory;

  // Create the tabs
  auto runsPresenter = createRunsTab(presenterFactory);
  auto settingsPresenter = createSettingsTab();

  // Create the presenter
  m_presenter.reset(
      new ReflMainWindowPresenter(this, runsPresenter, settingsPresenter));
}
```

Then:

```
IReflRunsTabPresenter *QtReflMainWindowView::createRunsTab(const ReflGenericDataProcessorPresenterFactory &factory) {

  QtReflRunsTabView *runsTab = new QtReflRunsTabView(this, factory);
  m_ui.mainTab->addTab(runsTab, QString("Runs"));

  return runsTab->getPresenter();
}
```

Why do it at this level and not at the `RunsTab` level?